docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+
+```sql
+CREATE STREAM s3 (
+    a VARCHAR KEY,
+    b INT
+) WITH (
+    kafka_topic = 's3',
+    partitions = 1,
+    value_format = 'avro'
+);
+```
+
+Insert some rows into the stream:
+
+```sql
+INSERT INTO s3 (a, b) VALUES ('k1', 3);
+INSERT INTO s3 (a, b) VALUES ('k1', 5);
+INSERT INTO s3 (a, b) VALUES ('k1', 7);
+INSERT INTO s3 (a, b) VALUES ('k2', 6);
+INSERT INTO s3 (a, b) VALUES ('k2', 2);
+```
+
+Execute a push query. Recall what `rolling_sum` does. It aggregates the previous three elements together, sums them up, and emits their output.
+
+```sql
+SELECT a, rolling_sum(b) AS MOVING_SUM FROM s3 GROUP BY a EMIT CHANGES;
+```
+
+Your output should look like the following. `k1` sums `3`, `5`, and `7` together to get a result of `15`. `k2` sums `6` and `2` together to get a result of `8`.
+
+```
++--------------------------------------------------------------+--------------------------------------------------------------+
+|A                                                             |MOVING_SUM                                                    |
++--------------------------------------------------------------+--------------------------------------------------------------+
+|k1                                                            |15                                                            |
+|k2                                                            |8                                                             |
+```
+
+Insert some more rows and shift older elements out of the aggregate:
+
+```sql
+INSERT INTO s3 (a, b) VALUES ('k1', 9);
+INSERT INTO s3 (a, b) VALUES ('k2', 1);
+INSERT INTO s3 (a, b) VALUES ('k2', 6);
+```
+
+Run the query again:
+
+```sql
+SELECT a, rolling_sum(b) AS MOVING_SUM FROM s3 GROUP BY a EMIT CHANGES;
+```
+
+And you should now see these results. In `k1`, the previous three values are now `5`, `7`, and `9`. In `k2`, the elements are `2`, `1`, and `6`.
+
+```
++--------------------------------------------------------------+--------------------------------------------------------------+
+|A                                                             |MOVING_SUM                                                    |
++--------------------------------------------------------------+--------------------------------------------------------------+
+|k1                                                            |21                                                            |
+|k2                                                            |9                                                             |
+```
+
+## Working with structs
+
+Using structs in UDFs requires somewhat more ceremony and deserves special attention. Structs are different from the other Java types that ksqlDB interfaces with because their typing is more dynamic. Fields can be added and removed, and their types are inferred on the fly. Because of this dynamism, UDFs need to be more explicit in their type contract with ksqlDB.
```

Comment:
```suggestion
Using structs in UDFs requires a more specific type contract with ksqlDB. Structs are different from the other Java types that ksqlDB interfaces with because their typing is more dynamic. Fields can be added and removed, and their types are inferred on the fly. Because of this dynamism, UDFs need to be more explicit in their type contract with ksqlDB.
```